OutOfSchool/OutOfSchool.AuthCommon/Services/Password/PasswordGenerator.cs

```diff
@@ -1,65 +1,50 @@
-﻿namespace OutOfSchool.AuthCommon.Services.Password;
+﻿using System.Security.Cryptography;
+
+namespace OutOfSchool.AuthCommon.Services.Password;
 
 public static class PasswordGenerator
 {
     /// <summary>
-    /// Generates a Random Password
-    /// respecting the given strength requirements.
+    /// Generates a Random Password.
     /// </summary>
-    /// <param name="passwordOptions">A valid PasswordOptions object
-    /// containing the password strength requirements.</param>
     /// <returns>A random password.</returns>
-    public static string GenerateRandomPassword(PasswordOptions passwordOptions)
+    public static string GenerateRandomPassword()
     {
-        string[] randomChars = new[]
-        {
-            "ABCDEFGHJKLMNOPQRSTUVWXYZ",    // uppercase
-            "abcdefghijkmnopqrstuvwxyz",    // lowercase
-            "0123456789",                   // digits
-            "!@$?_-",                       // non-alphanumeric
-        };
+        string[] allowedCharSets =
+        [
+            "ABCDEFGHJKLMNOPQRSTUVWXYZ",
+            "abcdefghijkmnopqrstuvwxyz",
+            "0123456789",
+            Constants.ValidationSymbols,
+        ];
 
-        CryptoRandom rand = new CryptoRandom();
+        List<char> password = [];
 
-        List<char> chars = new List<char>();
-
-        if (passwordOptions.RequireUppercase)
+        foreach (var charSet in allowedCharSets)
         {
-            chars.Insert(
-                rand.Next(0, chars.Count),
-                randomChars[0][rand.Next(0, randomChars[0].Length)]);
+            password.Insert(
+                GetRandomInt32(password.Count),
+                charSet[GetRandomInt32(charSet.Length)]);
         }
 
-        if (passwordOptions.RequireLowercase)
+        while (password.Count < Constants.PasswordMinLength)
         {
-            chars.Insert(
-                rand.Next(0, chars.Count),
-                randomChars[1][rand.Next(0, randomChars[1].Length)]);
+            var randomCharSetIndex = GetRandomInt32(allowedCharSets.Length);
+            var randomChar =
+                allowedCharSets[randomCharSetIndex][
+                GetRandomInt32(allowedCharSets[randomCharSetIndex].Length)];
+
+            if (!password.Contains(randomChar))
+            {
+                password.Add(randomChar);
+            }
```

Comment:
(Computational hardness) Please explain why we can't have duplicated symbols. Now you are loosing "randomness" here. 

Details. For case of `n` allowed chars for password having length `m`, where `m<n`, you schema provides `n!/(n-m)!` different passwords, while schema where duplications are allowed provides `n^m` paswords - the latter is the higher number.

Reply:
I was wrong in thinking that it would be better if the password didn't have duplicate symbols. Thank you!